### PR TITLE
Fix config use for block write and add IRQ and special mode checks

### PR DESCRIPTION
**Pull request by averycblack**

### CODE REVIEW

File: VoodooSMBus/VoodooSMBusControllerDriver.cpp [hunk 1]
```diff
@@ -263,8 +269,15 @@ IOWorkLoop* VoodooSMBusControllerDriver::getWorkLoop() {
 
 
 void VoodooSMBusControllerDriver::handleInterrupt(OSObject* owner, IOInterruptEventSource* src, int intCount) {
+    u16 pci_status = 0;
     u8 status;
-
+    
+    // Confirm this is our interrupt
+    pci_status = pci_device->configRead16(SMBPCISTS);
+    if (!(pci_status & SMBPCISTS_INTS)) {
```
Comment (leo-labs) [inline VoodooSMBus/VoodooSMBusControllerDriver.cpp:277]: Why is that necessary? I am just curious...

**Review by leo-labs — APPROVED** [review]

File: VoodooSMBus/VoodooSMBusControllerDriver.cpp (hunk 1 above)
Comment (averycblack) [reply VoodooSMBus/VoodooSMBusControllerDriver.cpp:277]: It was from i801.c, not exactly sure it's needed. I noticed it would fire once when it finished initializing but that's the only time I noticed it. We could remove it if we wanted to.

**Comment by averycblack** [comment]
This PR seems to cause issues on various devices, just gonna close it as VoodooSMBus works fine as it is. Maybe can revisit this if we find issues.
